napari/layers/utils/color_manager.py

```diff
@@ -0,0 +1,521 @@
+                         color_properties. This should be used for categorical
+                         properties only.
+     color_properties : Optional[ColorProperties]
+        The property values that are used for setting colors in ColorMode.COLORMAP
+        and ColorMode.CYCLE. The ColorProperties dataclass has 3 fields: name,
+        values, and current_value. name (str) is the name of the property being used.
+        values (np.ndarray) is an array containing the property values.
+        current_value contains the value for the next item to be added. color_properties
+        can be set as either a ColorProperties object or a dictionary where the keys are
+        the field values and the values are the field values (i.e., a dictionary that would
+        be valid in ColorProperties(**input_dictionary) ).
+    continuous_colormap : Colormap
+        The napari colormap object used in ColorMode.COLORMAP mode. This can also be set
+        using the name of a known colormap as a string.
+    contrast_limits : Tuple[float, float]
+        The min and max value for the colormap being applied to the color_properties
+        in ColorMonde.COLORMAP mode. Set as a tuple (min, max).
+    categorical_colormap : CategoricalColormap
+        The napari CategoricalColormap object used in ColorMode.CYCLE mode.
+        To set a direct mapping between color_property values and colors,
+        pass a dictionary where the keys are the property values and the
+        values are colors (either string names or (4,) color arrays).
+        To use a color cycle, pass a list or array of colors. You can also
+        pass the CategoricalColormap keyword arguments as a dictionary.
+    colors : np.ndarray
+        The colors in a Nx4 color array, where N is the number of colors.
+    """
+
+    # fields
+    current_color: Optional[Array[float, (4,)]] = None
+    mode: ColorMode = ColorMode.DIRECT
+    color_properties: Optional[ColorProperties] = None
+    continuous_colormap: Colormap = 'viridis'
+    contrast_limits: Optional[Tuple[float, float]] = None
+    categorical_colormap: CategoricalColormap = [0, 0, 0, 1]
+    colors: Array[float, (-1, 4)] = []
+
+    # validators
+    @validator('continuous_colormap', pre=True)
+    def _ensure_continuous_colormap(cls, v):
+        coerced_colormap = ensure_colormap(v)
+        return coerced_colormap
+
+    @validator('categorical_colormap', pre=True)
+    def _coerce_categorical_colormap(cls, v):
+        if isinstance(v, CategoricalColormap):
+            return v
+        if isinstance(v, list) or isinstance(v, np.ndarray):
+            return CategoricalColormap.from_array(v)
+        elif isinstance(v, dict):
+            return CategoricalColormap.from_dict(v)
+        else:
+            raise TypeError('colormap should be an array or dict')
+
+    @validator('color_properties', pre=True)
+    def _coerce_color_properties(cls, v):
```

Comment:
same here.  can this go on `ColorProperties`?

Reply:
Yup, done!